src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -187,8 +194,34 @@ export function fetchGenericAssayDataByStableIdsAndMolecularIds(
 
 export function makeGenericAssayOption(
     meta: GenericAssayMeta,
-    isPlotsTabOption?: boolean
+    isPlotsTabOption?: boolean,
```

Comment:
this function doesn't share much between "modes" isPlotsTabOption true/false.   Maybe just make it two functions one for plots, one for not plots?